### PR TITLE
[CI Example Analyzer] Fix some bugs

### PR DESCRIPTION
**Pull request by MoChilia**
## Description

- Fix false positive for Mismatched_Parameter_Value_Type
1. Support creating object by [Type]::new() 
2. Support creating object by type with its members like ```$bytes = [System.Text.Encoding]::Unicode.GetBytes($text)```
3. Support analyzing variables assigned by assignment statement like ```$VMName = 'v'+ $ResourceGroupName```
4. Support members from inherited interface
5. Support assigned by convert expression like ```[System.DayOfWeek[]]$WeekDays = @([System.DayOfWeek]::Monday..[System.DayOfWeek]::Friday)```
6. Support the situation when ActualType is subclass of ExpectedType 
7. Support Implicit type conversion
8. Support analyzing with "Where-Object" and "Sort-Object"
9. Skip "ForEach-Object" and "Select-Object"

- Fix false positive for Unbinded_Expression: Skip special situation of "Get-AzContext, Select-AzContext, Remove-AzContext" who have dynamic parameter "-Name"
- Skip the examples whose output can not be splitted from code
- No missingExample for autogenerated codes

TestPR: https://github.com/Azure/azure-powershell/pull/18872/

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: tools/StaticAnalysis/ExampleAnalyzer/utils.ps1 [hunk 1]
```diff
@@ -75,16 +75,21 @@ function Get-ExamplesDetailsFromMd {
     $indexOfExamples = $fileContent.IndexOf($EXAMPLES_HEADING)
     $indexOfParameters = $fileContent.IndexOf($PARAMETERS_HEADING)
 
-    $exampleNumber = 0
+    $exampleNumber = -1
```
Comment (BethanyZhou) [inline tools/StaticAnalysis/ExampleAnalyzer/utils.ps1:78]: why do you change the initial value as -1.

File: tools/StaticAnalysis/Exceptions/Az.Accounts/ExampleIssues.csv [hunk 2]
```diff
@@ -0,0 +1,8 @@
+"Module","Cmdlet","Example","RuleName","ProblemID","Severity","Description","Extent","Remediation"
+"Accounts","Disconnect-AzAccount","2","Unbinded_Expression","5014","2","Get-AzContext 'Work' is not explicitly assigned to a parameter.","'Work'","Assign 'Work' explicitly to the parameter."
```
Comment (BethanyZhou) [inline tools/StaticAnalysis/Exceptions/Az.Accounts/ExampleIssues.csv:2]: Why this file needs to be checked in?

File: tools/StaticAnalysis/ExampleAnalyzer/AnalyzeRules/ParameterNameAndValue.psm1 [hunk 3]
```diff
@@ -98,13 +120,27 @@ function Get-RecoveredValueType{
...
+                    for($i = 0; $i -lt $Type.ImplementedInterfaces.Length; $i++){
+                        $Member = $Type.ImplementedInterfaces[$i].GetMembers() | Where-Object {$_.Name -eq $Items[$j]}
+                        if($null -ne $Member){
+                            break
+                        }
+                    
+                }
+
+            }
+       if($null -eq $Member){
```
Comment (BethanyZhou) [inline tools/StaticAnalysis/ExampleAnalyzer/AnalyzeRules/ParameterNameAndValue.psm1:138]: Please correct the indent

**Comment by BethanyZhou** [comment]
Looks good to me. Any thoughts?

**Review by isra-fel — APPROVED** [review]

**Pull request merged**